database/postgresql_adapter.go

```diff
@@ -140,6 +143,8 @@ func (a *postgreSqlAdapter) copyHeaders(reader *csv.Reader, batchSize int, previ
 		return
 	}
 
+	bh := service.DefaultBlockHasher()
```

Comment:
Why define it here? I think it could be defined `calculateFields` method, where it's being used

Reply:
It looks like dependency, I would move it even higher